In [2]:
import pandas as pd
import time
import os
import numpy as np



In [127]:

countries_df = pd.read_csv("../Resources/DataFrames/All_olympic_competitors/athleteEventnocRegionsKEY.csv")

# Athlete Data
filepath1 = "../Resources/DataFrames/Cleaned/athleteEventsAggFinalNaN.csv"
final_athlete_df = pd.read_csv(filepath1)

# Cleaned World Data
# df444.to_csv('../Resources/DataFrames/world_data_cleaned.csv', index=False, header=True)
filepath2 = "../Resources/DataFrames/Cleaned/world_data_cleaned.csv"
cleaned_world = pd.read_csv(filepath2)


In [128]:
countries_df = countries_df.rename(columns = {"region": "Country Name"})
countries_df.tail(50)

NOC           Country Name                   notes
180  SLE           Sierra Leone                     NaN
181  SLO               Slovenia                     NaN
182  SMR             San Marino                     NaN
183  SOL        Solomon Islands                     NaN
184  SOM                Somalia                     NaN
185  SRB                 Serbia                     NaN
186  SRI              Sri Lanka                     NaN
187  SSD            South Sudan                     NaN
188  STP  Sao Tome and Principe                     NaN
189  SUD                  Sudan                     NaN
190  SUI            Switzerland                     NaN
191  SUR               Suriname                     NaN
192  SVK               Slovakia                     NaN
193  SWE                 Sweden                     NaN
194  SWZ              Swaziland                     NaN
195  SYR                  Syria                     NaN
196  TAN               Tanzania                     NaN
197  TCH         Czech Republic                     NaN
198  TGA                  Tonga                     NaN
199  THA               Thailand                     NaN
200  TJK             Tajikistan                     NaN
201  TKM           Turkmenistan                     NaN
202  TLS            Timor-Leste                     NaN
203  TOG                   Togo                     NaN
204  TPE                 Taiwan                     NaN
205  TTO               Trinidad     Trinidad and Tobago
206  TUN                Tunisia                     NaN
207  TUR                 Turkey                     NaN
208  TUV                    NaN                  Tuvalu
209  UAE   United Arab Emirates                     NaN
210  UAR                  Syria    United Arab Republic
211  UGA                 Uganda                     NaN
212  UKR                Ukraine                     NaN
213  UNK                    NaN                 Unknown
214  URS                 Russia                     NaN
215  URU                Uruguay                     NaN
216  USA                    USA                     NaN
217  UZB             Uzbekistan                     NaN
218  VAN                Vanuatu                     NaN
219  VEN              Venezuela                     NaN
220  VIE                Vietnam                     NaN
221  VIN          Saint Vincent                     NaN
222  VNM                Vietnam                     NaN
223  WIF               Trinidad  West Indies Federation
224  YAR                  Yemen             North Yemen
225  YEM                  Yemen                     NaN
226  YMD                  Yemen             South Yemen
227  YUG                 Serbia              Yugoslavia
228  ZAM                 Zambia                     NaN
229  ZIM               Zimbabwe                     NaN

In [129]:
print(final_athlete_df.shape)
final_athlete_df

(3310, 14)


Year  NOC   Country Name  Total Athletes  Medal Count  Gold Count  \
0     1896  AUS      Australia               3          2.0         2.0   
1     1896  AUT        Austria               3          2.0         2.0   
2     1896  DEN        Denmark               3          3.0         1.0   
3     1896  FRA         France              12          2.0         5.0   
4     1896  GBR             UK              14          4.0         4.0   
...    ...  ...            ...             ...          ...         ...   
3305  2016  VIE        Vietnam              22          0.0         1.0   
3306  2016  VIN  Saint Vincent               4          0.0         0.0   
3307  2016  YEM          Yemen               3          0.0         0.0   
3308  2016  ZAM         Zambia               7          0.0         0.0   
3309  2016  ZIM       Zimbabwe              30          0.0         0.0   

      Silver Count  Bronze Count  Average Age (M)  Average Height (M)  \
0              0.0           2.0        22.250000                 NaN   
1              1.0           2.0        21.750000                 NaN   
2              2.0           3.0        25.933333                 NaN   
3              4.0           2.0        19.789474                 NaN   
4              3.0           4.0        23.941176          188.000000   
...            ...           ...              ...                 ...   
3305           1.0           0.0        31.000000          170.000000   
3306           0.0           0.0        18.500000          173.000000   
3307           0.0           0.0        21.000000          171.500000   
3308           0.0           0.0        25.000000          174.750000   
3309           0.0           0.0        30.666667          175.333333   

      Average Weight (M)  Average Age (F)  Average Height (F)  \
0                    NaN              NaN                 NaN   
1                    NaN              NaN                 NaN   
2                    NaN              NaN                 NaN   
3                    NaN              NaN                 NaN   
4             102.000000              NaN                 NaN   
...                  ...              ...                 ...   
3305           67.416667        23.812500          164.875000   
3306           55.000000        23.000000          178.000000   
3307           68.500000        16.000000          166.000000   
3308           68.750000        22.000000          176.000000   
3309           71.666667        26.181818          164.772727   

      Average Weight (F)  
0                    NaN  
1                    NaN  
2                    NaN  
3                    NaN  
4                    NaN  
...                  ...  
3305           53.875000  
3306           65.000000  
3307           60.000000  
3308           65.000000  
3309           58.318182  

[3310 rows x 14 columns]

In [130]:
print(cleaned_world.shape)
cleaned_world

(5586, 68)


Country Name Country Code  \
0                     Afghanistan          AFG   
1     Africa Eastern and Southern          AFE   
2      Africa Western and Central          AFW   
3                         Albania          ALB   
4                         Algeria          DZA   
...                           ...          ...   
5581           West Bank and Gaza          PSE   
5582                        World          WLD   
5583                  Yemen, Rep.          YEM   
5584                       Zambia          ZMB   
5585                     Zimbabwe          ZWE   

                                            Series Name        Series Code  \
0       Armed forces personnel (% of total labor force)  MS.MIL.TOTL.TF.ZS   
1       Armed forces personnel (% of total labor force)  MS.MIL.TOTL.TF.ZS   
2       Armed forces personnel (% of total labor force)  MS.MIL.TOTL.TF.ZS   
3       Armed forces personnel (% of total labor force)  MS.MIL.TOTL.TF.ZS   
4       Armed forces personnel (% of total labor force)  MS.MIL.TOTL.TF.ZS   
...                                                 ...                ...   
5581  Unemployment, total (% of total labor force) (...     SL.UEM.TOTL.ZS   
5582  Unemployment, total (% of total labor force) (...     SL.UEM.TOTL.ZS   
5583  Unemployment, total (% of total labor force) (...     SL.UEM.TOTL.ZS   
5584  Unemployment, total (% of total labor force) (...     SL.UEM.TOTL.ZS   
5585  Unemployment, total (% of total labor force) (...     SL.UEM.TOTL.ZS   

     1960 1961 1962 1963 1964 1965  ...               2014               2015  \
0      ..   ..   ..   ..   ..   ..  ...   3.72705501582188   3.70073240669033   
1      ..   ..   ..   ..   ..   ..  ...   0.62903026918525  0.605281900453963   
2      ..   ..   ..   ..   ..   ..  ...  0.301808284295757   0.32145256580984   
3      ..   ..   ..   ..   ..   ..  ...  0.676692327901219  0.649676728503153   
4      ..   ..   ..   ..   ..   ..  ...    2.8122225290764     2.708967785061   
...   ...  ...  ...  ...  ...  ...  ...                ...                ...   
5581   ..   ..   ..   ..   ..   ..  ...             20.527             23.005   
5582   ..   ..   ..   ..   ..   ..  ...   6.01622857271861   6.04884902024336   
5583   ..   ..   ..   ..   ..   ..  ...             13.467               17.9   
5584   ..   ..   ..   ..   ..   ..  ...              8.131              8.316   
5585   ..   ..   ..   ..   ..   ..  ...              4.774              5.383   

                   2016               2017               2018  \
0      3.62353877713756   3.49015888651181    2.9525937221301   
1     0.538655325908266  0.514660592658541   0.51173656360947   
2     0.313934275271801  0.317382883149051  0.324083698170408   
3     0.627307198977415  0.653443392561926  0.638574587995223   
4      2.66136273518283   2.62652431622955   2.59293546960679   
...                 ...                ...                ...   
5581             23.939             25.677             26.256   
5582   6.01033134504685   5.91614141999205   5.75639790322219   
5583             18.382             18.564             17.644   
5584              8.418              8.518              5.031   
5585              5.893              6.347              6.792   

                   2019               2020              2021  \
0      3.01507875591777   1.81204522469526                ..   
1     0.497166127180999  0.435665062486187                ..   
2     0.382851379056212  0.329197509106008                ..   
3     0.558267806649528   0.58835815714458                ..   
4      2.55603419602847   2.74380149151702                ..   
...                 ...                ...               ...   
5581              25.34             25.895             26.39   
5582    5.5829801043763   6.58528753143071  6.03376129229279   
5583             17.303             17.972            18.248   
5584              5.539              6.032             5.195   
5585              7.373  

In [131]:
list(cleaned_world.columns)

['Country Name',
 'Country Code',
 'Series Name',
 'Series Code',
 '1960',
 '1961',
 '1962',
 '1963',
 '1964',
 '1965',
 '1966',
 '1967',
 '1968',
 '1969',
 '1970',
 '1971',
 '1972',
 '1973',
 '1974',
 '1975',
 '1976',
 '1977',
 '1978',
 '1979',
 '1980',
 '1981',
 '1982',
 '1983',
 '1984',
 '1985',
 '1986',
 '1987',
 '1988',
 '1989',
 '1990',
 '1991',
 '1992',
 '1993',
 '1994',
 '1995',
 '1996',
 '1997',
 '1998',
 '1999',
 '2000',
 '2001',
 '2002',
 '2003',
 '2004',
 '2005',
 '2006',
 '2007',
 '2008',
 '2009',
 '2010',
 '2011',
 '2012',
 '2013',
 '2014',
 '2015',
 '2016',
 '2017',
 '2018',
 '2019',
 '2020',
 '2021',
 '2022',
 '2023']

# Here is any formatting that needs to be done on 
## cleaned_world

In [132]:
# changing the ".." values to pd.NA, then making them numeric to match formatting
cleaned_world = cleaned_world.replace('..', pd.NA)
numeric_columns = cleaned_world.columns.drop(['Country Name', 'Country Code', 'Series Name', 'Series Code'])
cleaned_world[numeric_columns] = cleaned_world[numeric_columns].apply(pd.to_numeric, errors='coerce')
print(cleaned_world.shape)
cleaned_world.head(2)

(5586, 68)


Country Name Country Code  \
0                  Afghanistan          AFG   
1  Africa Eastern and Southern          AFE   

                                       Series Name        Series Code  1960  \
0  Armed forces personnel (% of total labor force)  MS.MIL.TOTL.TF.ZS   NaN   
1  Armed forces personnel (% of total labor force)  MS.MIL.TOTL.TF.ZS   NaN   

   1961  1962  1963  1964  1965  ...      2014      2015      2016      2017  \
0   NaN   NaN   NaN   NaN   NaN  ...  3.727055  3.700732  3.623539  3.490159   
1   NaN   NaN   NaN   NaN   NaN  ...  0.629030  0.605282  0.538655  0.514661   

       2018      2019      2020  2021  2022  2023  
0  2.952594  3.015079  1.812045   NaN   NaN   NaN  
1  0.511737  0.497166  0.435665   NaN   NaN   NaN  

[2 rows x 68 columns]

In [133]:
# match Country Code to NOC for merging
cleaned_world = cleaned_world.rename(columns = {"Country Code": "NOC"})
cleaned_world.head(2)

Country Name  NOC  \
0                  Afghanistan  AFG   
1  Africa Eastern and Southern  AFE   

                                       Series Name        Series Code  1960  \
0  Armed forces personnel (% of total labor force)  MS.MIL.TOTL.TF.ZS   NaN   
1  Armed forces personnel (% of total labor force)  MS.MIL.TOTL.TF.ZS   NaN   

   1961  1962  1963  1964  1965  ...      2014      2015      2016      2017  \
0   NaN   NaN   NaN   NaN   NaN  ...  3.727055  3.700732  3.623539  3.490159   
1   NaN   NaN   NaN   NaN   NaN  ...  0.629030  0.605282  0.538655  0.514661   

       2018      2019      2020  2021  2022  2023  
0  2.952594  3.015079  1.812045   NaN   NaN   NaN  
1  0.511737  0.497166  0.435665   NaN   NaN   NaN  

[2 rows x 68 columns]

In [134]:
cleaned_world.shape

(5586, 68)

In [135]:
# adding a copy for second melt/pivot
clean_world = cleaned_world.copy()
clean_world = clean_world.drop("Series Code", axis =1)
clean_world.shape

(5586, 67)

In [136]:
# now organizing this df to match the structure of final_athlete_df
# row = Index#	Country Name	Country Code	Year	Armed forces personnel (% of total labor force)......	Unemployment, total (% of total labor force)
# where the other columns are in list(olympic_year_world['Series Name'])


# pd.melt

using_melt2 = pd.melt(clean_world, 
                    id_vars=['Country Name', 'NOC', 'Series Name'], 
                    var_name='Year', 
                    value_name='Value')
# using_melt

all_years_formatted = using_melt2.pivot_table(index=['Country Name', 'Year', 'NOC'], 
                                       columns='Series Name', 
                                       values='Value').reset_index()
# all_years_formatted
all_years_formatted.columns.name = None
# all_years_formatted
all_years_formatted = all_years_formatted.sort_values(["Year", "NOC"])
all_years_formatted.reset_index(drop = True, inplace = True)
all_years_formatted

Country Name  Year  NOC  \
0                            Aruba  1960  ABW   
1      Africa Eastern and Southern  1960  AFE   
2                      Afghanistan  1960  AFG   
3       Africa Western and Central  1960  AFW   
4                           Angola  1960  AGO   
...                            ...   ...  ...   
16925                       Kosovo  2023  XKX   
16926                  Yemen, Rep.  2023  YEM   
16927                 South Africa  2023  ZAF   
16928                       Zambia  2023  ZMB   
16929                     Zimbabwe  2023  ZWE   

       Armed forces personnel (% of total labor force)  \
0                                                  NaN   
1                                                  NaN   
2                                                  NaN   
3                                                  NaN   
4                                                  NaN   
...                                                ...   
16925                                              NaN   
16926                                              NaN   
16927                                              NaN   
16928                                              NaN   
16929                                              NaN   

       Compulsory education, duration (years)  \
0                                         NaN   
1                                         NaN   
2                                         NaN   
3                                         NaN   
4                                         NaN   
...                                       ...   
16925                                     NaN   
16926                                     9.0   
16927                                     9.0   
16928                                     7.0   
16929                                     7.0   

       Control of Corruption: Percentile Rank  \
0                                         NaN   
1                                         NaN   
2                                         NaN   
3                                         NaN   
4                                         NaN   
...                                       ...   
16925                                     NaN   
16926                                     NaN   
16927                                     NaN   
16928                                     NaN   
16929                                     NaN   

       Current health expenditure (% of GDP)  \
0                                        NaN   
1                                        NaN   
2                                        NaN   
3                                        NaN   
4                                        NaN   
...                                      ...   
16925                                    NaN   
16926                                    NaN   
16927                                    NaN   
16928                                    NaN   
16929                                    NaN   

       Current health expenditure per capita (current US$)  \
0                                                    NaN     
1                                                    NaN     
2                                                    NaN     
3                                                    NaN     
4                                                    NaN     
...                                                  ...     
16925                                                NaN     
16926                                                NaN     
16927                                                NaN     
16928                                                NaN     
16929                                                NaN     

       Employment to population ratio, 15+, total (%) (modeled ILO estimate)  \
0                                                    NaN                       
1                                                    NaN                    

In [137]:
all_years_formatted["Year"].unique()

array(['1960', '1961', '1962', '1963', '1964', '1965', '1966', '1967',
       '1968', '1969', '1970', '1971', '1972', '1973', '1974', '1975',
       '1976', '1977', '1978', '1979', '1980', '1981', '1982', '1983',
       '1984', '1985', '1986', '1987', '1988', '1989', '1990', '1991',
       '1992', '1993', '1994', '1995', '1996', '1997', '1998', '1999',
       '2000', '2001', '2002', '2003', '2004', '2005', '2006', '2007',
       '2008', '2009', '2010', '2011', '2012', '2013', '2014', '2015',
       '2016', '2017', '2018', '2019', '2020', '2021', '2022', '2023'],
      dtype=object)

# bringing in more data 

### /Resources/Model_data


Battle_deaths_model.csv

Disaster_cleaned_model.csv

Emisions_cleaned_model.csv

Gender_wage_gap_model.csv

Long-term_unemployment_model.csv

Perceived_health_model.csv


In [138]:
# Battle deaths
filepatha = "../Resources/DataFrames/Model_prep/Battle_deaths_model.csv"
dfa = pd.read_csv(filepatha)

# Disaster data
filepathb = "../Resources/DataFrames/Model_prep/Disaster_cleaned_model.csv"
dfb = pd.read_csv(filepathb)

#Emissions data
filepathc = "../Resources/DataFrames/Model_prep/Emisions_cleaned_model.csv"
dfc = pd.read_csv(filepathc)

#gender wage gap data
filepathd = "../Resources/DataFrames/Model_prep/Gender_wage_gap_model.csv"
dfd = pd.read_csv(filepathd)

# long unemployment data
filepathe = "../Resources/DataFrames/Model_prep/Long-term_unemployment_model.csv"
dfe = pd.read_csv(filepathe)

# perceieved health data
filepathf = "../Resources/DataFrames/Model_prep/Perceived_health_model.csv"
dff = pd.read_csv(filepathf)

In [139]:
olympic_NOC_list = countries_df["NOC"].unique().tolist()

## dfa battle deaths 

In [140]:
print(dfa.shape)
dfa.head(1)

(9044, 5)


Country Name Country Code                               Series Name  \
0  Afghanistan          AFG  Battle-related deaths (number of people)   

         year_TB Battle_deaths  
0  1989 (YR1989)          5174

In [141]:
dfaa = dfa.copy()
dfaa["year_TB"] = dfaa["year_TB"].replace(
    r'^(\d{4}) \(YR\d{4}\)$',
    r'\1',
    regex=True
)
dfaa = dfaa.rename(columns = {"year_TB": "Year", 
                             "Battle_deaths": "Battle-related deaths (Total)",
                             "Country Code": "NOC"})
dfaa = dfaa.drop(columns = ["Series Name"])

print(list(dfaa['Year'].unique()))
dfaa.head(2)

['1989', '1990', '1991', '1992', '1993', '1994', '1995', '1996', '1997', '1998', '1999', '2000', '2001', '2002', '2003', '2004', '2005', '2006', '2007', '2008', '2009', '2010', '2011', '2012', '2013', '2014', '2015', '2016', '2017', '2018', '2019', '2020', '2021', '2022']


Country Name  NOC  Year Battle-related deaths (Total)
0  Afghanistan  AFG  1989                          5174
1  Afghanistan  AFG  1990                          1478

In [142]:
print(dfaa.shape)
dfaa.columns

(9044, 4)


Index(['Country Name', 'NOC', 'Year', 'Battle-related deaths (Total)'], dtype='object')

In [143]:
dfaa = dfaa[['Country Name', 'Year', 'NOC', 'Battle-related deaths (Total)']]
dfaa.head(2)

Country Name  Year  NOC Battle-related deaths (Total)
0  Afghanistan  1989  AFG                          5174
1  Afghanistan  1990  AFG                          1478

In [144]:
# Checking to see what is in dfaa but not in olympics data (according to "NOC")

#ALG should be in olympics but has a different NOC in dfaa ->below

excluded_nocs = set(dfaa['NOC']) - set(olympic_NOC_list)
print("Excluded NOCs:", excluded_nocs)
excluded_data = dfaa[dfaa["NOC"].isin(excluded_nocs)]
what_excluded = excluded_data[["NOC", "Country Name"]].drop_duplicates()
what_excluded

Excluded NOCs: {'ASM', 'ZMB', 'SAS', 'MAC', 'KNA', 'NAC', 'SDN', 'MNG', 'EAP', 'CEB', 'AFE', 'DEU', 'NCL', 'TON', 'VCT', 'TSS', 'CRI', 'SVN', 'EAR', 'TCA', 'GIB', 'WSM', 'PSE', 'MNA', 'LBN', 'BHR', 'TSA', 'ECA', 'IDN', 'IDX', 'MNP', 'IDA', 'PSS', 'ATG', 'IDB', 'ZWE', 'SSF', 'TCD', 'KHM', 'PYF', 'LMY', 'INX', 'SSA', 'MAF', 'LBY', 'TMN', 'ABW', 'BWA', 'MIC', 'TGO', 'CYM', 'KWT', 'FCS', 'LVA', 'EAS', 'IMN', 'TEA', 'HTI', 'SYC', 'GMB', 'TZA', 'MMR', 'GNQ', 'EUU', 'VUT', 'AFW', 'VGB', 'CSS', 'LDC', 'UMC', 'MCO', 'BGD', 'GTM', 'DZA', 'GRC', 'FJI', 'BFA', 'NLD', 'LKA', 'LIC', 'NER', 'MWI', 'PHL', 'SXM', 'LCN', 'COG', 'BHS', 'GIN', 'TEC', 'XKX', 'TLA', 'IRN', 'LMC', 'ARB', 'SLV', 'CHL', 'CHE', 'LTE', 'IBT', 'BMU', 'CUW', 'GNB', 'MUS', 'HND', 'PRY', 'DNK', 'SAU', 'LSO', 'URY', 'IBD', 'ECS', 'MEA', 'BTN', 'SGP', 'MDG', 'BLZ', 'PRT', 'SLB', 'PRE', 'FRO', 'PRI', 'ARE', 'VIR', 'HPC', 'PST', 'SST', 'NIC', 'OMN', 'OSS', 'WLD', 'AGO', 'HIC', 'BRB', 'NGA', 'ZAF', 'EMU', 'BGR', 'MRT', 'NPL', 'GRD', 'MYS

NOC                                Country Name
68    DZA                                     Algeria
102   ASM                              American Samoa
170   AGO                                      Angola
204   ATG                         Antigua and Barbuda
306   ABW                                       Aruba
...   ...                                         ...
8874  SSF                          Sub-Saharan Africa
8908  SSA  Sub-Saharan Africa (excluding high income)
8942  TSS   Sub-Saharan Africa (IDA & IBRD countries)
8976  UMC                         Upper middle income
9010  WLD                                       World

[145 rows x 2 columns]

In [145]:
dfaa = dfaa[dfaa["NOC"].isin(olympic_NOC_list)]
print(dfaa.shape)

(4114, 4)


## dfb disaster data

In [146]:
print(dfb.shape)
dfb.head(1)

(1820, 7)


Country  ISO  Start Year  AID Contribution ('000 US$)  Total Deaths  \
0  Canada  CAN        1912                            0            28   

   No. Injured  Total Damage, Adjusted ('000 US$)  
0          200                             157862

In [147]:
dfbb = dfb.copy()
dfbb = dfbb.rename(columns = {
    "Country": "Country Name", 
    "ISO": "NOC", 
    "Start Year": "Year", 
    "Total Deaths": "Total Disaster Related Deaths", 
    "No. Injured": "Total Disaster Related Injured", 
    "AID Contribution ('000 US$)": "Total Disaster Aid Contribution (adjusted US$)", 
    "Total Damage, Adjusted ('000 US$)": "Total Disaster Damage (adjusted US$)"
})

print(list(dfbb['Year'].unique()))
dfbb.head(2)

[1912, 1923, 1927, 1930, 1939, 1943, 1944, 1948, 1949, 1951, 1952, 1953, 1954, 1955, 1956, 1957, 1960, 1962, 1963, 1964, 1965, 1966, 1967, 1968, 1969, 1970, 1971, 1972, 1973, 1974, 1975, 1976, 1977, 1978, 1979, 1980, 1981, 1982, 1983, 1984, 1985, 1986, 1987, 1988, 1989, 1990, 1991, 1992, 1993, 1994, 1995, 1996, 1997, 1998, 1999, 2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018, 2019, 2020, 2021, 2022, 2023]


Country Name  NOC  Year  Total Disaster Aid Contribution (adjusted US$)  \
0       Canada  CAN  1912                                               0   
1        Japan  JPN  1923                                               0   

   Total Disaster Related Deaths  Total Disaster Related Injured  \
0                             28                             200   
1                         143000                          103733   

   Total Disaster Damage (adjusted US$)  
0                                157862  
1                              10692938

In [148]:
print(dfbb.shape)
dfbb.columns

(1820, 7)


Index(['Country Name', 'NOC', 'Year',
       'Total Disaster Aid Contribution (adjusted US$)',
       'Total Disaster Related Deaths', 'Total Disaster Related Injured',
       'Total Disaster Damage (adjusted US$)'],
      dtype='object')

In [149]:
dfbb = dfbb[['Country Name', 'Year', 'NOC', 'Total Disaster Aid Contribution (adjusted US$)',
       'Total Disaster Related Deaths', 'Total Disaster Related Injured',
       'Total Disaster Damage (adjusted US$)']]
dfbb.head(2)

Country Name  Year  NOC  Total Disaster Aid Contribution (adjusted US$)  \
0       Canada  1912  CAN                                               0   
1        Japan  1923  JPN                                               0   

   Total Disaster Related Deaths  Total Disaster Related Injured  \
0                             28                             200   
1                         143000                          103733   

   Total Disaster Damage (adjusted US$)  
0                                157862  
1                              10692938

## dfc Emissions data

In [150]:
print(dfc.shape)
dfc.head(1)

(11884, 4)


Reference area REF_AREA  TIME_PERIOD  total emissions by tonnes
0    Netherlands      NLD         1990                   346.5895

In [151]:
dfcc = dfc.copy()
dfcc = dfcc.rename(columns = {
    "Reference area": "Country Name", 
    "REF_AREA": "NOC", 
    "TIME_PERIOD": "Year", 
    "total emissions by tonnes": "Total Dangerous Emissions (tonnes)"
})

print(list(dfcc['Year'].unique()))
dfcc.head(2)

[1990, 1991, 1992, 1993, 1994, 1995, 1996, 1997, 1998, 1999, 2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018, 2019, 2020, 2021, 2022]


Country Name  NOC  Year  Total Dangerous Emissions (tonnes)
0  Netherlands  NLD  1990                            346.5895
1  Netherlands  NLD  1991                            360.3044

In [152]:
print(dfcc.shape)
dfcc.columns

(11884, 4)


Index(['Country Name', 'NOC', 'Year', 'Total Dangerous Emissions (tonnes)'], dtype='object')

In [153]:
dfcc = dfcc[['Country Name', 'Year', 'NOC', 'Total Dangerous Emissions (tonnes)']]
dfcc.head(2)

Country Name  Year  NOC  Total Dangerous Emissions (tonnes)
0  Netherlands  1990  NLD                            346.5895
1  Netherlands  1991  NLD                            360.3044

## dfd gender wage gap

In [154]:
print(dfd.shape)
dfd.head(1)

(947, 4)


Reference area REF_AREA  TIME_PERIOD  Median_%_of_men_wages_gap
0      Australia      AUS         1980                      18.75

In [155]:
dfdd = dfd.copy()
dfdd = dfdd.rename(columns = {
    "Reference area": "Country Name", 
    "REF_AREA": "NOC", 
    "TIME_PERIOD": "Year", 
    "Median_%_of_men_wages_gap": "Median % Wage Gap (% of Male Wage)"
})

print(list(dfdd['Year'].unique()))
dfdd.head(2)

[1980, 1979, 1978, 2016, 2015, 2014, 2019, 2018, 2017, 2013, 2012, 2011, 2010, 2009, 2004, 2003, 2002, 2001, 2000, 1999, 1998, 1997, 1995, 1994, 1993, 1992, 1991, 1990, 1989, 1988, 1977, 1987, 1986, 2022, 1982, 1981, 2005, 2021, 2020, 2007, 2008, 2006, 1976, 1975, 2023, 1985, 1984, 1983, 1996, 1974, 1973, 1972, 1971, 1970]


Country Name  NOC  Year  Median % Wage Gap (% of Male Wage)
0    Australia  AUS  1980                               18.75
1    Australia  AUS  1979                               20.00

In [156]:
print(dfdd.shape)
dfdd.columns

(947, 4)


Index(['Country Name', 'NOC', 'Year', 'Median % Wage Gap (% of Male Wage)'], dtype='object')

In [157]:
dfdd = dfdd[['Country Name', 'Year', 'NOC', 'Median % Wage Gap (% of Male Wage)']]
dfdd.head(2)

Country Name  Year  NOC  Median % Wage Gap (% of Male Wage)
0    Australia  1980  AUS                               18.75
1    Australia  1979  AUS                               20.00

## dfe long-term unemployment

In [158]:
print(dfe.shape)
dfe.head(1)

(1326, 5)


REF_AREA Reference area          Age  TIME_PERIOD  \
0      GRC         Greece  Middle-aged         2004   

   Long-term_unemployment_%_of_labor  
0                           5.202932

In [159]:
dfe['Age'].unique()

array(['Middle-aged', 'Young'], dtype=object)

In [160]:
dfee = dfe.copy()
dfee = dfee.rename(columns = {
    "Reference area": "Country Name", 
    "REF_AREA": "NOC", 
    "TIME_PERIOD": "Year", 
    "Long-term_unemployment_%_of_labor": "x" 
})

print(list(dfee['Year'].unique()))
dfee.head(2)

[2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018, 2019, 2020, 2021, 2022]


NOC Country Name          Age  Year         x
0  GRC       Greece  Middle-aged  2004  5.202932
1  GRC       Greece  Middle-aged  2005  4.958358

In [161]:
#split up young and middle-aged
young = dfee[dfee['Age'] == 'Young']
middle = dfee[dfee['Age'] == 'Middle-aged']
young = young.rename(columns = {"x": "Long-term Unemployment (% of labor - YOUNG)"})
middle = middle.rename(columns = {"x": "Long-term Unemployment (% of labor - MIDDLE-AGE)"})
young = young.drop('Age', axis=1)
middle = middle.drop('Age', axis=1)
both = young.merge(middle, on = ["Country Name", "NOC", "Year"], how = "outer")
dfeee = both.copy()
dfeee

NOC Country Name  Year  Long-term Unemployment (% of labor - YOUNG)  \
0    AUT      Austria  2004                                     1.814354   
1    AUT      Austria  2005                                     1.461086   
2    AUT      Austria  2006                                     1.532547   
3    AUT      Austria  2007                                     1.264225   
4    AUT      Austria  2008                                     1.166080   
..   ...          ...   ...                                          ...   
659  POL       Poland  2020                                     1.638286   
660  POL       Poland  2021                                     1.807343   
661  POL       Poland  2022                                     2.083962   
662  KOR        Korea  2010                                          NaN   
663  KOR        Korea  2012                                          NaN   

     Long-term Unemployment (% of labor - MIDDLE-AGE)  
0                                            1.423493  
1                                            1.363981  
2                                            1.373698  
3                                            1.281036  
4                                            0.959454  
..                                                ...  
659                                          0.566186  
660                                          0.851908  
661                                          0.813670  
662                                          0.014973  
663                                          0.012268  

[664 rows x 5 columns]

In [162]:
# some data does not match. ie:
countries_df[countries_df["Country Name"] == "South Korea"]

NOC Country Name notes
109  KOR  South Korea   NaN

In [163]:
print(dfeee.shape)
dfeee.columns

(664, 5)


Index(['NOC', 'Country Name', 'Year',
       'Long-term Unemployment (% of labor - YOUNG)',
       'Long-term Unemployment (% of labor - MIDDLE-AGE)'],
      dtype='object')

In [164]:
dfeee = dfeee[['Country Name', 'Year', 'NOC', 'Long-term Unemployment (% of labor - YOUNG)',
       'Long-term Unemployment (% of labor - MIDDLE-AGE)']]
dfeee.head(2)

Country Name  Year  NOC  Long-term Unemployment (% of labor - YOUNG)  \
0      Austria  2004  AUT                                     1.814354   
1      Austria  2005  AUT                                     1.461086   

   Long-term Unemployment (% of labor - MIDDLE-AGE)  
0                                          1.423493  
1                                          1.363981

## dff preceived health (self)

In [165]:
print(dff.shape)
dff.head(1)

(1443, 6)


STRUCTURE_NAME REF_AREA Reference area  TIME_PERIOD                  Age  \
0  Health status      FIN        Finland         1980  From 25 to 44 years   

   % of pop perceived   
0                 73.5

In [166]:
print(list(dff['Age'].unique()))

['From 25 to 44 years', 'From 15 to 24 years']


In [167]:
dff.columns

Index(['STRUCTURE_NAME', 'REF_AREA', 'Reference area', 'TIME_PERIOD', 'Age',
       '% of pop perceived '],
      dtype='object')

In [168]:
dfff = dff.copy()
dfff = dfff.drop("STRUCTURE_NAME", axis = 1)
dfff = dfff.rename(columns = {
    "Reference area": "Country Name", 
    "REF_AREA": "NOC", 
    "TIME_PERIOD": "Year", 
    "% of pop perceived ": "x" 
})

print(list(dfff['Year'].unique()))
dfff.head(2)

[1980, 1981, 1982, 1983, 1984, 1985, 1986, 1987, 1988, 1989, 1990, 1991, 1992, 1993, 1994, 1995, 1996, 1997, 1998, 1999, 2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018, 2019, 2020, 2021, 2022, 2023]


NOC Country Name  Year                  Age     x
0  FIN      Finland  1980  From 25 to 44 years  73.5
1  FIN      Finland  1981  From 25 to 44 years  75.4

In [169]:
# adjust 'age' to young and middle-aged
dfff['Age'] = dfff['Age'].replace({
    "From 25 to 44 years": "Middle-aged",
    "From 15 to 24 years": "Young"
})
dfff.head(1)

NOC Country Name  Year          Age     x
0  FIN      Finland  1980  Middle-aged  73.5

In [170]:
#split up young and middle-aged
young2 = dfff[dfff['Age'] == 'Young']
middle2 = dfff[dfff['Age'] == 'Middle-aged']
young2 = young2.rename(columns = {"x": "% Population Perveived Healthy(self) - YOUNG"})
middle2 = middle2.rename(columns = {"x": "% Population Perveived Healthy(self) - MIDDLE-AGE"})
young2 = young2.drop('Age', axis=1)
middle2 = middle2.drop('Age', axis=1)
both2 = young2.merge(middle2, on = ["Country Name", "NOC", "Year"], how = "outer")
dffff = both2.copy()
dffff

NOC Country Name  Year  % Population Perveived Healthy(self) - YOUNG  \
0    ROU      Romania  2008                                          96.8   
1    ROU      Romania  2009                                          97.6   
2    ROU      Romania  2010                                          97.6   
3    ROU      Romania  2011                                          97.3   
4    ROU      Romania  2012                                          96.3   
..   ...          ...   ...                                           ...   
829  HRV      Croatia  2019                                          95.3   
830  HRV      Croatia  2020                                          95.8   
831  HRV      Croatia  2021                                          95.9   
832  HRV      Croatia  2022                                          96.1   
833  HRV      Croatia  2023                                          96.7   

     % Population Perveived Healthy(self) - MIDDLE-AGE  
0                                                 91.2  
1                                                 92.1  
2                                                 92.7  
3                                                 91.6  
4                                                 92.9  
..                                                 ...  
829                                                NaN  
830                                                NaN  
831                                                NaN  
832                                                NaN  
833                                                NaN  

[834 rows x 5 columns]

In [171]:
print(dffff.shape)
dffff.columns

(834, 5)


Index(['NOC', 'Country Name', 'Year',
       '% Population Perveived Healthy(self) - YOUNG',
       '% Population Perveived Healthy(self) - MIDDLE-AGE'],
      dtype='object')

In [172]:
dffff = dffff[['Country Name', 'Year', 'NOC', '% Population Perveived Healthy(self) - YOUNG',
       '% Population Perveived Healthy(self) - MIDDLE-AGE']]
dffff.head(2)

Country Name  Year  NOC  % Population Perveived Healthy(self) - YOUNG  \
0      Romania  2008  ROU                                          96.8   
1      Romania  2009  ROU                                          97.6   

   % Population Perveived Healthy(self) - MIDDLE-AGE  
0                                               91.2  
1                                               92.1

## Merge 6 new dfs onto combined

dfaa

dfbb

dfcc

dfdd

dfeee

dffff

* first onto cleaned_world - has all years
* will need to filt

In [173]:
all_years_formatted


Country Name  Year  NOC  \
0                            Aruba  1960  ABW   
1      Africa Eastern and Southern  1960  AFE   
2                      Afghanistan  1960  AFG   
3       Africa Western and Central  1960  AFW   
4                           Angola  1960  AGO   
...                            ...   ...  ...   
16925                       Kosovo  2023  XKX   
16926                  Yemen, Rep.  2023  YEM   
16927                 South Africa  2023  ZAF   
16928                       Zambia  2023  ZMB   
16929                     Zimbabwe  2023  ZWE   

       Armed forces personnel (% of total labor force)  \
0                                                  NaN   
1                                                  NaN   
2                                                  NaN   
3                                                  NaN   
4                                                  NaN   
...                                                ...   
16925                                              NaN   
16926                                              NaN   
16927                                              NaN   
16928                                              NaN   
16929                                              NaN   

       Compulsory education, duration (years)  \
0                                         NaN   
1                                         NaN   
2                                         NaN   
3                                         NaN   
4                                         NaN   
...                                       ...   
16925                                     NaN   
16926                                     9.0   
16927                                     9.0   
16928                                     7.0   
16929                                     7.0   

       Control of Corruption: Percentile Rank  \
0                                         NaN   
1                                         NaN   
2                                         NaN   
3                                         NaN   
4                                         NaN   
...                                       ...   
16925                                     NaN   
16926                                     NaN   
16927                                     NaN   
16928                                     NaN   
16929                                     NaN   

       Current health expenditure (% of GDP)  \
0                                        NaN   
1                                        NaN   
2                                        NaN   
3                                        NaN   
4                                        NaN   
...                                      ...   
16925                                    NaN   
16926                                    NaN   
16927                                    NaN   
16928                                    NaN   
16929                                    NaN   

       Current health expenditure per capita (current US$)  \
0                                                    NaN     
1                                                    NaN     
2                                                    NaN     
3                                                    NaN     
4                                                    NaN     
...                                                  ...     
16925                                                NaN     
16926                                                NaN     
16927                                                NaN     
16928                                                NaN     
16929                                                NaN     

       Employment to population ratio, 15+, total (%) (modeled ILO estimate)  \
0                                                    NaN                       
1                                                    NaN                    

In [174]:
all_years_formatted["Year"].unique()

array(['1960', '1961', '1962', '1963', '1964', '1965', '1966', '1967',
       '1968', '1969', '1970', '1971', '1972', '1973', '1974', '1975',
       '1976', '1977', '1978', '1979', '1980', '1981', '1982', '1983',
       '1984', '1985', '1986', '1987', '1988', '1989', '1990', '1991',
       '1992', '1993', '1994', '1995', '1996', '1997', '1998', '1999',
       '2000', '2001', '2002', '2003', '2004', '2005', '2006', '2007',
       '2008', '2009', '2010', '2011', '2012', '2013', '2014', '2015',
       '2016', '2017', '2018', '2019', '2020', '2021', '2022', '2023'],
      dtype=object)

In [175]:
all_years_formatted['Year'].astype(int)
dfaa['Year'].astype(int)
dfbb['Year'].astype(int)
dfcc['Year'].astype(int)
dfdd['Year'].astype(int)
dfeee['Year'].astype(int)
dffff['Year'].astype(int)

0      2008
1      2009
2      2010
3      2011
4      2012
       ... 
829    2019
830    2020
831    2021
832    2022
833    2023
Name: Year, Length: 834, dtype: int64

In [176]:
all_years_formatted.shape

(16930, 24)

In [177]:
merging1 = all_years_formatted.merge(dfaa, on = ["Country Name", "NOC", "Year"], how = "outer")
merging1['Year'] = merging1['Year'].astype(int)
print(merging1.shape)

(16930, 25)


In [178]:
merging2 = merging1.merge(dfbb, on = ["Country Name", "NOC", "Year"], how = "outer")
merging2['Year'] = merging2['Year'].astype(int)
print(merging2.shape)

(17932, 29)


In [179]:
merging3 = merging2.merge(dfcc, on = ["Country Name", "NOC", "Year"], how = "outer")
merging3['Year'] = merging3['Year'].astype(int)
print(merging3.shape)

(28469, 30)


In [180]:
merging4 = merging3.merge(dfdd, on = ["Country Name", "NOC", "Year"], how = "outer")
merging4['Year'] = merging4['Year'].astype(int)
print(merging4.shape)

(28521, 31)


In [181]:
merging5 = merging4.merge(dfeee, on = ["Country Name", "NOC", "Year"], how = "outer")
merging5['Year'] = merging5['Year'].astype(int)
print(merging5.shape)

(28521, 33)


In [182]:
merging6 = merging5.merge(dffff, on = ["Country Name", "NOC", "Year"], how = "outer")
merging6['Year'] = merging6['Year'].astype(int)
print(merging6.shape)

(28523, 35)


# Here is formatting, filtering, aggregating, and matching for Country Name and NOC

Only want to include countries who have olympic data in the countries_df


steps to do:
* If there is a country with the wrong country code (algeria = ALG in countries_df but DRZ in other dfs) adjust (if country name is in countries_df['Country Name'], change the NOC to match the NOC in countries_df). ie: All Algeria rows should now have NOC == 'ALG'
* merge and aggregate data (average any duplicate countries (korea/south korea) on countries_df['NOC'] to average sereis data (198 & NaN) onto one row (with the correct NOC)
* Adjust Country Name to correct name from countries_df['Country Name'] based on merging6['NOC']==countries_df['NOC']
* filter merging6 to only include data for countries that are in countries_df (for only countries who have been to an olympics)
* adjust any missing data ('..', or something else) to be NaN
* will NaN +value (198) /2 = 198? will aggregating ignore NaN values??
* save
* filter for only olympic years
* save

In [183]:
countries_df.shape

(230, 3)

In [184]:
merging6.columns

Index(['Country Name', 'Year', 'NOC',
       'Armed forces personnel (% of total labor force)',
       'Compulsory education, duration (years)',
       'Control of Corruption: Percentile Rank',
       'Current health expenditure (% of GDP)',
       'Current health expenditure per capita (current US$)',
       'Employment to population ratio, 15+, total (%) (modeled ILO estimate)',
       'GDP (current US$)', 'GDP growth (annual %)',
       'GDP per capita (current US$)', 'GDP per capita growth (annual %)',
       'Government Effectiveness: Percentile Rank',
       'Government expenditure on education, total (% of GDP)',
       'International tourism, expenditures (current US$)',
       'Military expenditure (% of GDP)',
       'PM2.5 air pollution, population exposed to levels exceeding WHO guideline value (% of total)',
       'Political Stability and Absence of Violence/Terrorism: Percentile Rank',
       'Population growth (annual %)', 'Population, total',
       'Rule of Law: Perce

In [185]:
NOC_dict = dict(zip(countries_df["Country Name"], countries_df["NOC"]))

In [186]:
# update NOC codes based on Country Name
for index, row in merging6.iterrows():
    if row["Country Name"] in NOC_dict:
        merging6.at[index, "NOC"] = NOC_dict[row["Country Name"]]

In [187]:
merging6["Battle-related deaths (Total)"] = merging6["Battle-related deaths (Total)"].apply(pd.to_numeric, errors='coerce')
merging6["Battle-related deaths (Total)"].astype(float)
print(merging6.dtypes)

Country Name                                                                                     object
Year                                                                                              int64
NOC                                                                                              object
Armed forces personnel (% of total labor force)                                                 float64
Compulsory education, duration (years)                                                          float64
Control of Corruption: Percentile Rank                                                          float64
Current health expenditure (% of GDP)                                                           float64
Current health expenditure per capita (current US$)                                             float64
Employment to population ratio, 15+, total (%) (modeled ILO estimate)                           float64
GDP (current US$)                                               

In [188]:
# aggregate by the updated NOC
merging66 = merging6.groupby(["NOC", "Year"]).agg({
    'Country Name': "first",
   'Armed forces personnel (% of total labor force)': "mean",
   'Compulsory education, duration (years)': "mean",
   'Control of Corruption: Percentile Rank': "mean",
   'Current health expenditure (% of GDP)': "mean",
   'Current health expenditure per capita (current US$)': "mean",
   'Employment to population ratio, 15+, total (%) (modeled ILO estimate)': "mean",
   'GDP (current US$)': "mean",
    'GDP growth (annual %)': "mean",
   'GDP per capita (current US$)': "mean",
    'GDP per capita growth (annual %)': "mean",
   'Government Effectiveness: Percentile Rank': "mean",
   'Government expenditure on education, total (% of GDP)': "mean",
   'International tourism, expenditures (current US$)': "mean",
   'Military expenditure (% of GDP)': "mean",
   'PM2.5 air pollution, population exposed to levels exceeding WHO guideline value (% of total)': "mean",
   'Political Stability and Absence of Violence/Terrorism: Percentile Rank': "mean",
   'Population growth (annual %)': "mean",
    'Population, total': "mean",
   'Rule of Law: Percentile Rank': "mean",
   'Strength of legal rights index (0=weak to 12=strong)': "mean",
   'Unemployment, total (% of total labor force) (modeled ILO estimate)': "mean",
   'Battle-related deaths (Total)': "mean",
   'Total Disaster Aid Contribution (adjusted US$)': "mean",
   'Total Disaster Related Deaths': "mean",
    'Total Disaster Related Injured': "mean",
   'Total Disaster Damage (adjusted US$)': "mean",
   'Total Dangerous Emissions (tonnes)': "mean",
   'Median % Wage Gap (% of Male Wage)': "mean",
   'Long-term Unemployment (% of labor - YOUNG)': "mean",
   'Long-term Unemployment (% of labor - MIDDLE-AGE)': "mean",
   '% Population Perveived Healthy(self) - YOUNG': "mean",
   '% Population Perveived Healthy(self) - MIDDLE-AGE': "mean"
}).reset_index()

In [189]:
# update country names by NOC
NOC_to_country = dict(zip(countries_df["NOC"], countries_df["Country Name"]))
# update NOC codes based on Country Name
for index, row in merging66.iterrows():
    if row["NOC"] in NOC_to_country:
        merging66.at[index, "Country Name"] = NOC_to_country[row["NOC"]]

In [190]:
#adjusting country names
name_changes = {
    "Boliva": "Bolivia",
    "Ivory Coast": "Côte d'Ivoire",
    "Trinidad": "Trinidad and Tobago",
    "South Korea": "Korea",
    "North Korea": "Korea, Democratic People's Republic of"
}

merging66['Country Name'] = merging66['Country Name'].replace(name_changes)

In [191]:
# Only include olympic countries
merging66 = merging66[merging66["NOC"].isin(countries_df["NOC"])]

In [192]:
print(merging66.shape)

(12393, 35)


In [193]:
len(merging66["Country Name"].unique())

194

In [194]:
merging66["Country Name"].unique()

array(['Afghanistan', 'Curacao', 'Albania', 'Algeria', 'Andorra',
       'Angola', 'Argentina', 'Armenia', 'Aruba', 'American Samoa',
       'Australia', 'Austria', 'Azerbaijan', 'Bahamas', 'Bangladesh',
       'Barbados', 'Burundi', 'Belgium', 'Benin', 'Bermuda', 'Bhutan',
       'Bosnia and Herzegovina', 'Belize', 'Belarus', 'Bolivia',
       'Botswana', 'Brazil', 'Bahrain', 'Bulgaria', 'Burkina Faso',
       'Central African Republic', 'Cambodia', 'Cayman Islands', 'Chad',
       'Chile', "Côte d'Ivoire", 'Cameroon',
       'Democratic Republic of the Congo', 'Colombia', 'Comoros',
       'Cape Verde', 'Costa Rica', 'Croatia', 'Cuba', 'Cyprus',
       'Czech Republic', 'Denmark', 'Djibouti', 'Dominica',
       'Dominican Republic', 'Ecuador', 'Egypt', 'Eritrea', 'El Salvador',
       'Spain', 'Estonia', 'Ethiopia', 'Fiji', 'Finland', 'France',
       'Micronesia', 'Gabon', 'UK', 'Guinea-Bissau', 'Georgia',
       'Equatorial Guinea', 'Ghana', 'Greece', 'Grenada', 'Guatemala',
      

In [195]:
merging66_countries = set(merging66['Country Name'])
countries_df_countries = set(countries_df['Country Name'])

print("Countries in merging66 but not in countries_df:")
print(merging66_countries - countries_df_countries)
print()
print("Countries in countries_df but not in merging66:")
print(countries_df_countries - merging66_countries)

Countries in merging66 but not in countries_df:
{'Korea', 'Trinidad and Tobago', "Côte d'Ivoire", "Korea, Democratic People's Republic of", 'Bolivia'}

Countries in countries_df but not in merging66:
{'Virgin Islands, US', 'Saint Kitts', 'Antigua', 'Iran', 'North Korea', 'Boliva', 'Taiwan', 'Virgin Islands, British', 'Palestine', 'Gambia', 'Trinidad', 'Brunei', 'Republic of Congo', 'Ivory Coast', 'Individual Olympic Athletes', 'South Korea', 'Saint Vincent', 'Cook Islands'}


In [196]:
#saving

# merging66.to_csv('../Resources/DataFrames/Cleaned/all_merged_countries_and_series.csv', index=False, header=True)
merged_all = pd.read_csv('../Resources/DataFrames/Cleaned/all_merged_countries_and_series.csv')
print(merged_all.shape)
merged_all.head(2)


(12393, 35)


NOC  Year Country Name  Armed forces personnel (% of total labor force)  \
0  AFG  1956  Afghanistan                                              NaN   
1  AFG  1960  Afghanistan                                              NaN   

   Compulsory education, duration (years)  \
0                                     NaN   
1                                     NaN   

   Control of Corruption: Percentile Rank  \
0                                     NaN   
1                                     NaN   

   Current health expenditure (% of GDP)  \
0                                    NaN   
1                                    NaN   

   Current health expenditure per capita (current US$)  \
0                                                NaN     
1                                                NaN     

   Employment to population ratio, 15+, total (%) (modeled ILO estimate)  \
0                                                NaN                       
1                                                NaN                       

   GDP (current US$)  ...  Total Disaster Aid Contribution (adjusted US$)  \
0                NaN  ...                                             0.0   
1                NaN  ...                                             NaN   

   Total Disaster Related Deaths  Total Disaster Related Injured  \
0                          100.0                          2000.0   
1                            NaN                             NaN   

   Total Disaster Damage (adjusted US$)  Total Dangerous Emissions (tonnes)  \
0                              280228.0                                 NaN   
1                                   NaN                                 NaN   

   Median % Wage Gap (% of Male Wage)  \
0                                 NaN   
1                                 NaN   

   Long-term Unemployment (% of labor - YOUNG)  \
0                                          NaN   
1                                          NaN   

   Long-term Unemployment (% of labor - MIDDLE-AGE)  \
0                                               NaN   
1                                               NaN   

   % Population Perveived Healthy(self) - YOUNG  \
0                                           NaN   
1                                           NaN   

   % Population Perveived Healthy(self) - MIDDLE-AGE  
0                                                NaN  
1                                                NaN  

[2 rows x 35 columns]

In [197]:
# adjusting the df for only olympic years
years_olympic = final_athlete_df["Year"].unique()
years_olympic = list(years_olympic)

print(years_olympic)



[1896, 1900, 1904, 1906, 1908, 1912, 1920, 1924, 1928, 1932, 1936, 1948, 1952, 1956, 1960, 1964, 1968, 1972, 1976, 1980, 1984, 1988, 1992, 1994, 1996, 1998, 2000, 2002, 2004, 2006, 2008, 2010, 2012, 2014, 2016]


In [198]:
olympic_year_world = merged_all[merged_all["Year"].isin(years_olympic)]

In [199]:
print(olympic_year_world.shape)
olympic_year_world

(4067, 35)


NOC  Year Country Name  \
0      AFG  1956  Afghanistan   
1      AFG  1960  Afghanistan   
5      AFG  1964  Afghanistan   
9      AFG  1968  Afghanistan   
13     AFG  1972  Afghanistan   
...    ...   ...          ...   
12377  ZIM  2008     Zimbabwe   
12379  ZIM  2010     Zimbabwe   
12381  ZIM  2012     Zimbabwe   
12383  ZIM  2014     Zimbabwe   
12385  ZIM  2016     Zimbabwe   

       Armed forces personnel (% of total labor force)  \
0                                                  NaN   
1                                                  NaN   
5                                                  NaN   
9                                                  NaN   
13                                                 NaN   
...                                                ...   
12377                                         1.082556   
12379                                         1.061204   
12381                                         1.030244   
12383                                         0.979296   
12385                                         0.927032   

       Compulsory education, duration (years)  \
0                                         NaN   
1                                         NaN   
5                                         NaN   
9                                         NaN   
13                                        NaN   
...                                       ...   
12377                                     7.0   
12379                                     7.0   
12381                                     7.0   
12383                                     7.0   
12385                                     7.0   

       Control of Corruption: Percentile Rank  \
0                                         NaN   
1                                         NaN   
5                                         NaN   
9                                         NaN   
13                                        NaN   
...                                       ...   
12377                                3.883495   
12379                                2.857143   
12381                                1.895735   
12383                                4.326923   
12385                                9.523809   

       Current health expenditure (% of GDP)  \
0                                        NaN   
1                                        NaN   
5                                        NaN   
9                                        NaN   
13                                       NaN   
...                                      ...   
12377                                    NaN   
12379                              10.475839   
12381                               6.917949   
12383                               8.132273   
12385                               7.466447   

       Current health expenditure per capita (current US$)  \
0                                                    NaN     
1                                                    NaN     
5                                                    NaN     
9                                                    NaN     
13                                                   NaN     
...                                                  ...     
12377                                                NaN     
12379                                          98.249458     
12381                                          89.260956     
12383                                         114.444298     
12385                                         106.189697     

       Employment to population ratio, 15+, total (%) (modeled ILO estimate)  \
0                                                    NaN                       
1                                                    NaN                       
5                                                    NaN                       
9                                                    NaN                       
13           

In [200]:
#save
# olympic_year_world.to_csv('../Resources/DataFrames/Cleaned/all_merged_countries_and_series_OLYMPICYEARS.csv', index=False, header=True)
merged_all_olympic_only = pd.read_csv('../Resources/DataFrames/Cleaned/all_merged_countries_and_series_OLYMPICYEARS.csv')
print(merged_all_olympic_only.shape)
merged_all_olympic_only.head(2)


(4067, 35)


NOC  Year Country Name  Armed forces personnel (% of total labor force)  \
0  AFG  1956  Afghanistan                                              NaN   
1  AFG  1960  Afghanistan                                              NaN   

   Compulsory education, duration (years)  \
0                                     NaN   
1                                     NaN   

   Control of Corruption: Percentile Rank  \
0                                     NaN   
1                                     NaN   

   Current health expenditure (% of GDP)  \
0                                    NaN   
1                                    NaN   

   Current health expenditure per capita (current US$)  \
0                                                NaN     
1                                                NaN     

   Employment to population ratio, 15+, total (%) (modeled ILO estimate)  \
0                                                NaN                       
1                                                NaN                       

   GDP (current US$)  ...  Total Disaster Aid Contribution (adjusted US$)  \
0                NaN  ...                                             0.0   
1                NaN  ...                                             NaN   

   Total Disaster Related Deaths  Total Disaster Related Injured  \
0                          100.0                          2000.0   
1                            NaN                             NaN   

   Total Disaster Damage (adjusted US$)  Total Dangerous Emissions (tonnes)  \
0                              280228.0                                 NaN   
1                                   NaN                                 NaN   

   Median % Wage Gap (% of Male Wage)  \
0                                 NaN   
1                                 NaN   

   Long-term Unemployment (% of labor - YOUNG)  \
0                                          NaN   
1                                          NaN   

   Long-term Unemployment (% of labor - MIDDLE-AGE)  \
0                                               NaN   
1                                               NaN   

   % Population Perveived Healthy(self) - YOUNG  \
0                                           NaN   
1                                           NaN   

   % Population Perveived Healthy(self) - MIDDLE-AGE  
0                                                NaN  
1                                                NaN  

[2 rows x 35 columns]

In [208]:
all_all = pd.merge(final_athlete_df, olympic_year_world, 
                     on=['Year', 'NOC', 'Country Name'], 
                     how='outer')
print(all_all.shape)
all_all

(4985, 46)


Year  NOC Country Name  Total Athletes  Medal Count  Gold Count  \
0     1896  AUS    Australia             3.0          2.0         2.0   
1     1896  AUT      Austria             3.0          2.0         2.0   
2     1896  DEN      Denmark             3.0          3.0         1.0   
3     1896  FRA       France            12.0          2.0         5.0   
4     1896  GBR           UK            14.0          4.0         4.0   
...    ...  ...          ...             ...          ...         ...   
4980  1994  ZIM     Zimbabwe             NaN          NaN         NaN   
4981  1998  ZIM     Zimbabwe             NaN          NaN         NaN   
4982  2002  ZIM     Zimbabwe             NaN          NaN         NaN   
4983  2006  ZIM     Zimbabwe             NaN          NaN         NaN   
4984  2010  ZIM     Zimbabwe             NaN          NaN         NaN   

      Silver Count  Bronze Count  Average Age (M)  Average Height (M)  ...  \
0              0.0           2.0        22.250000                 NaN  ...   
1              1.0           2.0        21.750000                 NaN  ...   
2              2.0           3.0        25.933333                 NaN  ...   
3              4.0           2.0        19.789474                 NaN  ...   
4              3.0           4.0        23.941176               188.0  ...   
...            ...           ...              ...                 ...  ...   
4980           NaN           NaN              NaN                 NaN  ...   
4981           NaN           NaN              NaN                 NaN  ...   
4982           NaN           NaN              NaN                 NaN  ...   
4983           NaN           NaN              NaN                 NaN  ...   
4984           NaN           NaN              NaN                 NaN  ...   

      Total Disaster Aid Contribution (adjusted US$)  \
0                                                NaN   
1                                                NaN   
2                                                NaN   
3                                                NaN   
4                                                NaN   
...                                              ...   
4980                                             NaN   
4981                                             NaN   
4982                                             NaN   
4983                                             NaN   
4984                                             NaN   

      Total Disaster Related Deaths  Total Disaster Related Injured  \
0                               NaN                             NaN   
1                               NaN                             NaN   
2                               NaN                             NaN   
3                               NaN                             NaN   
4                               NaN                             NaN   
...                             ...                             ...   
4980                            NaN                             NaN   
4981                            NaN                             NaN   
4982                            NaN                             NaN   
4983                            NaN                             NaN   
4984                            NaN                             NaN   

      Total Disaster Damage (adjusted US$)  \
0                                      NaN   
1                                      NaN   
2                                      NaN   
3                                      NaN   
4                                      NaN   
...                                    ...   
4980                                   NaN   
4981                                   NaN   
4982                                   NaN   
4983                                   NaN   
4984                                   NaN   

      Total Dangerous Emissions (tonnes)  Median % Wage Gap (% of Male Wage)  \
0                                

In [209]:
all_all.columns

Index(['Year', 'NOC', 'Country Name', 'Total Athletes', 'Medal Count',
       'Gold Count', 'Silver Count', 'Bronze Count', 'Average Age (M)',
       'Average Height (M)', 'Average Weight (M)', 'Average Age (F)',
       'Average Height (F)', 'Average Weight (F)',
       'Armed forces personnel (% of total labor force)',
       'Compulsory education, duration (years)',
       'Control of Corruption: Percentile Rank',
       'Current health expenditure (% of GDP)',
       'Current health expenditure per capita (current US$)',
       'Employment to population ratio, 15+, total (%) (modeled ILO estimate)',
       'GDP (current US$)', 'GDP growth (annual %)',
       'GDP per capita (current US$)', 'GDP per capita growth (annual %)',
       'Government Effectiveness: Percentile Rank',
       'Government expenditure on education, total (% of GDP)',
       'International tourism, expenditures (current US$)',
       'Military expenditure (% of GDP)',
       'PM2.5 air pollution, population expo

In [210]:
all_all = all_all[['NOC', 'Year', 'Country Name', 'Total Athletes', 'Medal Count', 
        'Gold Count', 'Silver Count', 'Bronze Count','Average Age (M)',
       'Average Height (M)', 'Average Weight (M)', 'Average Age (F)',
       'Average Height (F)', 'Average Weight (F)',
       'Armed forces personnel (% of total labor force)',
       'Compulsory education, duration (years)',
       'Control of Corruption: Percentile Rank',
       'Current health expenditure (% of GDP)',
       'Current health expenditure per capita (current US$)',
       'Employment to population ratio, 15+, total (%) (modeled ILO estimate)',
       'GDP (current US$)', 'GDP growth (annual %)',
       'GDP per capita (current US$)', 'GDP per capita growth (annual %)',
       'Government Effectiveness: Percentile Rank',
       'Government expenditure on education, total (% of GDP)',
       'International tourism, expenditures (current US$)',
       'Military expenditure (% of GDP)',
       'PM2.5 air pollution, population exposed to levels exceeding WHO guideline value (% of total)',
       'Political Stability and Absence of Violence/Terrorism: Percentile Rank',
       'Population growth (annual %)', 'Population, total',
       'Rule of Law: Percentile Rank',
       'Strength of legal rights index (0=weak to 12=strong)',
       'Unemployment, total (% of total labor force) (modeled ILO estimate)',
       'Battle-related deaths (Total)',
       'Total Disaster Aid Contribution (adjusted US$)',
       'Total Disaster Related Deaths', 'Total Disaster Related Injured',
       'Total Disaster Damage (adjusted US$)',
       'Total Dangerous Emissions (tonnes)',
       'Median % Wage Gap (% of Male Wage)',
       'Long-term Unemployment (% of labor - YOUNG)',
       'Long-term Unemployment (% of labor - MIDDLE-AGE)',
       '% Population Perveived Healthy(self) - YOUNG',
       '% Population Perveived Healthy(self) - MIDDLE-AGE']]

In [211]:
print(all_all.shape)
all_all

(4985, 46)


NOC  Year Country Name  Total Athletes  Medal Count  Gold Count  \
0     AUS  1896    Australia             3.0          2.0         2.0   
1     AUT  1896      Austria             3.0          2.0         2.0   
2     DEN  1896      Denmark             3.0          3.0         1.0   
3     FRA  1896       France            12.0          2.0         5.0   
4     GBR  1896           UK            14.0          4.0         4.0   
...   ...   ...          ...             ...          ...         ...   
4980  ZIM  1994     Zimbabwe             NaN          NaN         NaN   
4981  ZIM  1998     Zimbabwe             NaN          NaN         NaN   
4982  ZIM  2002     Zimbabwe             NaN          NaN         NaN   
4983  ZIM  2006     Zimbabwe             NaN          NaN         NaN   
4984  ZIM  2010     Zimbabwe             NaN          NaN         NaN   

      Silver Count  Bronze Count  Average Age (M)  Average Height (M)  ...  \
0              0.0           2.0        22.250000                 NaN  ...   
1              1.0           2.0        21.750000                 NaN  ...   
2              2.0           3.0        25.933333                 NaN  ...   
3              4.0           2.0        19.789474                 NaN  ...   
4              3.0           4.0        23.941176               188.0  ...   
...            ...           ...              ...                 ...  ...   
4980           NaN           NaN              NaN                 NaN  ...   
4981           NaN           NaN              NaN                 NaN  ...   
4982           NaN           NaN              NaN                 NaN  ...   
4983           NaN           NaN              NaN                 NaN  ...   
4984           NaN           NaN              NaN                 NaN  ...   

      Total Disaster Aid Contribution (adjusted US$)  \
0                                                NaN   
1                                                NaN   
2                                                NaN   
3                                                NaN   
4                                                NaN   
...                                              ...   
4980                                             NaN   
4981                                             NaN   
4982                                             NaN   
4983                                             NaN   
4984                                             NaN   

      Total Disaster Related Deaths  Total Disaster Related Injured  \
0                               NaN                             NaN   
1                               NaN                             NaN   
2                               NaN                             NaN   
3                               NaN                             NaN   
4                               NaN                             NaN   
...                             ...                             ...   
4980                            NaN                             NaN   
4981                            NaN                             NaN   
4982                            NaN                             NaN   
4983                            NaN                             NaN   
4984                            NaN                             NaN   

      Total Disaster Damage (adjusted US$)  \
0                                      NaN   
1                                      NaN   
2                                      NaN   
3                                      NaN   
4                                      NaN   
...                                    ...   
4980                                   NaN   
4981                                   NaN   
4982                                   NaN   
4983                                   NaN   
4984                                   NaN   

      Total Dangerous Emissions (tonnes)  Median % Wage Gap (% of Male Wage)  \
0                                

In [4]:
#save
# all_all.to_csv('../Resources/DataFrames/Cleaned/completed_dataframe_all_olympic_years.csv', index=False, header=True)
olympic_years_FINAL = pd.read_csv('../Resources/DataFrames/Cleaned/completed_dataframe_all_olympic_years.csv')
print(olympic_years_FINAL.shape)
olympic_years_FINAL.head(50)


(4985, 46)


NOC  Year    Country Name  Total Athletes  Medal Count  Gold Count  \
0   AUS  1896       Australia             3.0          2.0         2.0   
1   AUT  1896         Austria             3.0          2.0         2.0   
2   DEN  1896         Denmark             3.0          3.0         1.0   
3   FRA  1896          France            12.0          2.0         5.0   
4   GBR  1896              UK            14.0          4.0         4.0   
5   GER  1896         Germany            21.0          2.0        26.0   
6   GRE  1896          Greece           109.0         20.0        10.0   
7   HUN  1896         Hungary             7.0          3.0         2.0   
8   ITA  1896           Italy             1.0          0.0         0.0   
9   SUI  1896     Switzerland             3.0          0.0         1.0   
10  SWE  1896          Sweden             1.0          0.0         0.0   
11  USA  1896             USA            14.0          2.0        11.0   
12  ARG  1900       Argentina             1.0          0.0         0.0   
13  AUS  1900       Australia             7.0          3.0         7.0   
14  AUT  1900         Austria            17.0          3.0         0.0   
15  BEL  1900         Belgium            74.0         15.0         6.0   
16  BOH  1900  Czech Republic             9.0          3.0         0.0   
17  BRA  1900          Brazil             1.0          0.0         0.0   
18  CAN  1900          Canada             4.0          1.0         1.0   
19  COL  1900        Colombia            11.0          0.0         0.0   
20  CUB  1900            Cuba             1.0          0.0         1.0   
21  DEN  1900         Denmark            17.0          2.0         7.0   
22  ESP  1900           Spain            10.0          0.0         2.0   
23  FRA  1900          France           878.0         84.0        56.0   
24  GBR  1900              UK           174.0         36.0        84.0   
25  GER  1900         Germany            83.0         11.0        14.0   
26  GRE  1900          Greece            12.0          0.0         0.0   
27  HAI  1900           Haiti            31.0          0.0        17.0   
28  HUN  1900         Hungary            18.0          2.0         1.0   
29  IND  1900           India             1.0          0.0         0.0   
30  IRI  1900            Iran             1.0          0.0         0.0   
31  ITA  1900           Italy            23.0          0.0         3.0   
32  LUX  1900      Luxembourg             1.0          0.0         1.0   
33  MEX  1900          Mexico             4.0          4.0         0.0   
34  NED  1900     Netherlands            45.0         26.0         3.0   
35  NOR  1900          Norway             7.0          3.0         0.0   
36  NZL  1900     New Zealand             7.0          0.0         7.0   
37  PER  1900            Peru             1.0          0.0         0.0   
38  ROU  1900         Romania             1.0          0.0         0.0   
39  RUS  1900          Russia             5.0          0.0         0.0   
40  SUI  1900     Switzerland            18.0          1.0        16.0   
41  SWE  1900          Sweden            16.0          1.0         6.0   
42  USA  1900             USA           118.0         21.0        49.0   
43  AUS  1904       Australia             2.0          1.0         0.0   
44  AUT  1904         Austria             8.0          1.0         7.0   
45  CAN  1904          Canada            56.0         12.0        27.0   
46  CUB  1904            Cuba             3.0          0.0         5.0   
47  FRA  1904          France             6.0          0.0         0.0   
48  GBR  1904              UK             6.0          0.0         1.0   
49  GER  1904         Germany            41.0         11.0         4.0   

    Silver Count  Bronze Count  Average Age (M)  Average Height (M)  ...  \
0            0.0           2.0        22.250000                 NaN  ...   
1            1.0           2.0        21.750000                 NaN  ...   
2

In [12]:
x = olympic_years_FINAL[(olympic_years_FINAL['Year']==2000) & (olympic_years_FINAL['NOC']=="AZE")]

In [13]:
x.head(15)

NOC  Year Country Name  Total Athletes  Medal Count  Gold Count  \
1978  AZE  2000   Azerbaijan            31.0          1.0         2.0   

      Silver Count  Bronze Count  Average Age (M)  Average Height (M)  ...  \
1978           0.0           1.0             24.6             173.375  ...   

      Total Disaster Aid Contribution (adjusted US$)  \
1978                                           341.0   

      Total Disaster Related Deaths  Total Disaster Related Injured  \
1978                           31.0                           600.0   

      Total Disaster Damage (adjusted US$)  \
1978                               17695.0   

      Total Dangerous Emissions (tonnes)  Median % Wage Gap (% of Male Wage)  \
1978                           24.175382                                 NaN   

      Long-term Unemployment (% of labor - YOUNG)  \
1978                                          NaN   

      Long-term Unemployment (% of labor - MIDDLE-AGE)  \
1978                                               NaN   

      % Population Perveived Healthy(self) - YOUNG  \
1978                                           NaN   

      % Population Perveived Healthy(self) - MIDDLE-AGE  
1978                                                NaN  

[1 rows x 46 columns]

In [14]:
olympic_years_FINAL.shape

(4985, 46)